DashAI/back/dataloaders/classes/dataloader.py

```diff
@@ -0,0 +1,70 @@
+from abc import abstractmethod
+
+from datasets import ClassLabel
+
+from DashAI.back.config_object import ConfigObject
+
+
+class DataLoader(ConfigObject):
+    """
+    Abstract class with base methods for all data loaders
+    """
+
+    @abstractmethod
+    def load_data(self, dataset_path, url=None):
+        pass
+
+    @abstractmethod
+    def set_task_format(self, task):
+        pass
+
+    def split_dataset(self, dataset, params, class_column):
+        """
+        Split the dataset in train, test and validation data
+        """
```

Comment:
As a public method, this function should have good documentation and hopefully examples and typing.

```suggestion
    def split_dataset(self, dataset: TODO, params: dict[str, Any], class_column: str) -> TODO:
        """Split the dataset in train, test and validation data

        Parameters
        ----------
        dataset : TODO
            ...
        params : dict[str, Any]
            Dict of parameters that configure the splitting process.
            These can be: 
            - stratify: bool = True indicates that the split will be stratified. 
            - test_size: float = TODO
            - val_size: float = TODO
            By default TODO
        class_column : str
            In the case of stratifying, indicate the column with which to stratify.

        Returns
        -------
        TODO
            The splitted dataset.
        """
```

Reply:
In addition, I have some conceptual questions about this function: 

1. Why the parameters are passed in params and class_column (which is also parameter) independent of the dictionary. Moreover, is it necessary to have the parameters in params? Why you simply don't receive multiple parameters (something like def split_dataset(dataset, train_size, val_size, stratify, class_column, ...). In this case, you can make explicit which parameters the function receives (something that is not intuitive using a dictionary) and the types of each parameter.
2. I think it would be more correct that the function will make explicit that the operation sought is to generate three partitions: train test split. what if the user does not want to generate val and only use train and test?
3. What happens with the datasets that come previously separated?

Reply:
You're right is necessary a good documentation, I'll leave all the documentation in this branch tomorrow. Now in the branch front/dataloaders are more documentation.

Answering the questions:
1- The parameters are passed in this dictionary because in the endpoint this is obtained directly from the json received, so the 'splits' key is passed directly. But it effectively doesn't make sense to have this and separate the class column. I agree that it is better to make the parameters for this method explicit, I will change it.

2- From what was discussed in meetings I assumed that the three partitions had to exist, but it was thought that I could choose not to have a split by giving it a size of 0. Now that I see it better, this is not working with a partition at 0, but I can correct it to let him do it.

3- If the splits are defined before, this method is not called. This depend of a parameter ('folder_split' ir now renamed as 'splits_in_folders') that is checked in the endpoint.

Reply:
Thanks for the answers!

Regarding point 2, if you think that the behavior when giving 0 in some split is wrong, then at the beginning verify that the parameter is greater than 0 and in the case that it is not, raise an exception. 

```python
if not isinstance(val_size, float):
    raise TypeError("val_size muest be float, got {type(val_size)}") .

if val_size < 0 or val_size > 1:
    raise ValueError("val_size must be in the (0, 1) range (0 and 1 not included), got {val_size}")
```

Reply:
Thanks for your advice! I understand, I will clarify it in the code.